include/libtorrent/aux_/unique_ptr.hpp

```diff
@@ -50,6 +50,9 @@ namespace libtorrent { namespace aux {
 		using base = std::unique_ptr<T[]>;
 		using underlying_index = typename underlying_index_t<IndexType>::type;
 
+		unique_ptr() {}
+		explicit unique_ptr(T arr[]) : base(arr) {}
```

Comment:
I think it's a bit misleading to use the ``T arr[]`` syntax. It means the same as ``T* arr``, but that might not be obvious to everyone